File: backend/app/schemas/food.py
from pydantic import BaseModel, Field


class FoodRecognizeResult(BaseModel):
    food_name: str
    confidence: float
    calories: float
    protein: float
    fat: float
    carbohydrates: float
    serving_size: str


class FoodRecognizeResponse(BaseModel):
    success: bool
    detail: str
    items: list[FoodRecognizeResult] = []


class FoodRecordCreate(BaseModel):
    food_name: str = Field(..., max_length=100)
    meal_type: str = Field(default="lunch", pattern="^(breakfast|lunch|dinner|snack)$")
    calories: float = Field(..., ge=0)
    protein: float = Field(..., ge=0)
    fat: float = Field(..., ge=0)
    carbohydrates: float = Field(..., ge=0)
    serving_size: str | None = None
    image_url: str | None = None


class FoodSearchQuery(BaseModel):
    keyword: str = Field(..., min_length=1, max_length=50)

File: frontend/src/types/food.ts
export interface FoodRecord {
  id: number
  user_id: number
  food_name: string
  meal_type: 'breakfast' | 'lunch' | 'dinner' | 'snack'
  image_url: string | null
  calories: number
  protein: number
  fat: number
  carbohydrates: number
  serving_size: string | null
  recorded_at: string
}

export interface FoodRecognizeResult {
  food_name: string
  calories: number
  protein: number
  fat: number
  carbohydrates: number
  serving_size: string
  confidence: number
}

export interface NutritionLog {
  id: number
  user_id: number
  log_date: string
  total_calories: number
  total_protein: number
  total_fat: number
  total_carbohydrates: number
}
